amr-wind/equation_systems/vof/vof_advection.H

```diff
@@ -59,16 +62,59 @@ struct AdvectionOp<VOF, fvm::Godunov>
             for (amrex::MFIter mfi(dof_field(lev), mfi_info); mfi.isValid();
                  ++mfi) {
                 const auto& bx = mfi.tilebox();
-                amrex::FArrayBox tmpfab(amrex::grow(bx, 1), 3 * VOF::ndim + 2);
+                amrex::FArrayBox tmpfab(amrex::grow(bx, 1), 2 * VOF::ndim + 2);
                 tmpfab.setVal<amrex::RunOn::Device>(0.0);
-                multiphase::split_advection(
-                    lev, bx, isweep, dof_field(lev).array(mfi),
-                    u_mac(lev).const_array(mfi), v_mac(lev).const_array(mfi),
-                    w_mac(lev).const_array(mfi), aa_x(lev).array(mfi),
-                    aa_y(lev).array(mfi), aa_z(lev).array(mfi),
-                    dof_field.bcrec_device().data(), tmpfab.dataPtr(), geom, dt,
+
+                multiphase::cmask_loop(
+                    bx, dof_field(lev).array(mfi), (*fluxC)(lev).array(mfi),
                     m_use_lagrangian);
 
+                multiphase::split_advection_step(
```

Comment:
I am sure this can be done in a more efficient (and prettier) way, but I am happy for time being. I guess we can refactor later on